### PR TITLE
feat: css parser

### PR DESCRIPTION
**Pull request by naomi-lgbt**
Checklist:

- [X] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [X] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [X] My pull request targets the `main` branch of freeCodeCamp.
- [X] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #XXXXX

Load a css parser into the curriculum helpers for use in the tests.

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -63,6 +63,7 @@
     "chai": "4.3.4",
     "crypto-browserify": "3.12.0",
     "csrf": "3.1.0",
+    "css-tree": "^1.1.3",
```
Comment (ojeytonwilliams) [inline client/package.json:66]:
```suggestion
    "css-tree": "1.1.3",
```
To avoid upsetting the bot.

**Comment by Sembauke** [comment]
Will this replace RegEx in the future/current challenges?

**Comment by nhcarrigan** [comment]
That's the plan, yeah - to be able to use a parser to read the CSS instead of regexing it.

**Comment by raisedadead** [comment]
Could you add a sample change that uses the parser? Or point me to some snippet to help understand the process?

**Comment by nhcarrigan** [comment]
Sure thing - I'll push up a sample test change today

**Comment by nhcarrigan** [comment]
It looks like Shaun's got a better fix for this one - closing in favour of #42148

**Pull request closed without merging**